assets/mariadbapp/main.go

```diff
@@ -0,0 +1,89 @@
+/*
+   Copyright 2020 SUSE
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/cloudfoundry-community/go-cfenv"
+	_ "github.com/go-sql-driver/mysql"
+)
+
+var serviceName string
+
+func init() {
+	serviceName = os.Getenv("SERVICE_NAME")
+}
+
+func main() {
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	appEnv, err := cfenv.Current()
+	if err != nil {
+		log.Fatal(err)
+	}
+	mariadb, err := appEnv.Services.WithName(serviceName)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	uri, err := url.Parse(mariadb.Credentials["uri"].(string))
+	if err != nil {
+		log.Fatal(err)
+	}
+	uriStr := fmt.Sprintf("%s@tcp(%s)/%s", uri.User.String(), uri.Hostname(), strings.TrimPrefix(uri.Path, "/"))
```

Comment:
```suggestion
	config := mysql.NewConfig()
	config.User = uri.User.String()
	config.Net = "tcp"
	config.Addr = uri.Hostname()
	config.DBName = uri.Path
	uriStr := config.FormatDSN()
```
Or something along those lines, to not need to build strings manually (which can be rather error prone).